### DISCUSSION TITLE
Interpolate Dataset Name in Test Name

### DISCUSSION THREAD

**Discussion by JHWelch** [pestphp/pest#1143] [Ideas]
I like to be able to have my test names be as concise and descriptive as possible. I appreciate that we can name dataset entries, which makes them much more descriptive.

I would like to be able to take this a step further and be able to interpolate the name of the dataset somewhere in the test name so that it can be even more descriptive in plain english. 

## Example

With the following test for sorting inside of a table
```php
it('can sort by all fields', function ($field, $data, $expected) {
    /* ... */
})->with([
    'name' => [ /* ... */ ],
    'email' => [ /* ... */ ],
    'created_at' => [ /* ... */]
]);
```
The Pest response will be 

```log
   PASS  Tests\Feature\TableSortTest
  ✓ it can sort by all fields with dataset "name"                                                                                                            
  ✓ it can sort by all fields with dataset "email"
  ✓ it can sort by all fields with dataset "created_at"
```

I want to be able to update the name of the test with some sort of flag, similar to the `:attribute` placeholder in [Laravel validation](https://laravel.com/docs/11.x/validation)

So lets say I can update the test name to 
```php
it('can sort by :dataset field', function (/* ... */){/* ... */})
```

And then the tests would appear as 
```log
   PASS  Tests\Feature\TableSortTest
  ✓ it can sort by "name" field                                                                                                     
  ✓ it can sort by "email" field
  ✓ it can sort by "created_at" field
```

Or maybe without the `"`s auto applied. 

## Questions
- Is this possible or straightforward?
- Would others find this useful? 
- I would be interested in PRing this if feasible, does anyone know where in codebase to get started?

**Comment by fabio-ivona** [comment]
Like the idea, don't have the source code ad hand, but I think DatasetRepository is the place to look at ;)

**Comment by JHWelch** [comment]
Okay gave it a stab in #1146

**Comment by JHWelch** [comment]
Resolved in #1146
